### DISCUSSION TITLE
Dynamic fields in stubs

### DISCUSSION THREAD

**Discussion by kiranpatel11** [specmatic/specmatic#708] [Ideas and Suggestions]
As mentioned in the issue #705, 
I propose to consider the support for dynamic fields in the stubs.

For example, 
1. Mapping some of the fields from request to the corresponding response so that the users don't need to create and maintain 10s or 100s of stubs just to ensure such Ids are correlated for request/response. This is a very common usecase and supported by most established SV frameworks - microks, wiremock, mock-server, pact, karate etc.
e.g. Given a GET /pets/{petId} operation, when petId=1, the response should assign the id field with the same value to be useful in functional validation.
2. For fields such as created_timestamp, last_modified_timestamp in the response payload, usually it is expected to return the timestamp of the executed request. How to do that using specmatic SV ?
3. How to generate the fields with typical fields like valid email, US SSN number, postcodes, phone numbers ? can you integrate with the popular faker library so that users don't have to deal with complexity of the regex for typical usecases?

PS: created this discussion thread for gathering different views.

**Comment by harikrishnan83** [comment]
@kiranpatel11 Thanks for reaching out. We do have a mechanism that allows you to incorporate values from the request into your response. Here is the [documentation](https://specmatic.in/documentation/service_virtualization_tutorial.html#externalised-response-generation). Please see if this helps.
